flink-runtime/src/main/java/org/apache/flink/runtime/io/network/buffer/LocalBufferPool.java

```diff
@@ -463,4 +516,29 @@ private void returnExcessMemorySegments() {
 			returnMemorySegment(segment);
 		}
 	}
+
+	@Nullable
+	@Override
+	public BufferRecycler[] getSubpartitionBufferRecyclers() {
+		return subpartitionBufferRecyclers;
+	}
+
+	/**
+	 *
+	 */
+	class SubpartitionBufferRecycler implements BufferRecycler {
+
+		private int channel;
+		private LocalBufferPool bufferPool;
```

Comment:
As this class is not static, you don't need `bufferPool` field. So you can either drop this field, or make the `SubpartitionBufferRecycler` class static.